### ISSUE TITLE
(Unwanted/Unexpected) Colour Change on Brightness Adjustment

### ISSUE THREAD

**Issue by idofr** [esphome/issues#1611]
**Operating environment/Installation (Hass.io/Docker/pip/etc.):
Hass.io

**ESP (ESP32/ESP8266, Board/Sonoff):**
ESP32 (esp-wrover-kit)

**ESPHome version (latest production, beta, dev branch)**
1.15.3

**Affected component:**
https://esphome.io/components/light/neopixelbus.html

**Description of problem:**
So this is a bit of an off one.
I've configured my Philips Hue Switch via Zigbee2mqtt to control both my Hue lights as well as my EspHome lights. The configuration was done using NodeRed and is rather straight forward (see attachment). On should turn all lights on while alternative between predefined scenes. Up and down should dimm the lights and "off" should... well.. turn them off.
The funny thing is - when the "up" and "down" buttons are pressed, the Hue lights just dimm accordingly while the EspHome lights also alternate back and forth with the scenes.

![image](https://user-images.githubusercontent.com/2036083/98484365-25dc9900-220f-11eb-83c0-b7ed24a0c649.png)

Now as you can see my the attached NodeRed scheme. I'm also printing everything to the console for debugging. No debugging log line shows anything to do with colour/scene change (for the "up" and "down" buttons). I've then took a further step and have looked at the Esp logs.
These also don't show any evidence of a colour change.
For example.
When pressing "on" (which *should* alternate between scenes), I see the following log entries:
```
[21:51:47][D][light:265]:  'Right Lamp - Main' Setting:
[21:51:47][D][light:278]:   Brightness: 100%
[21:51:47][D][light:287]:   Red=38%, Green=33%, Blue=100%
[21:51:47][D][light:290]:   White Value: 35%
[21:51:47][D][light:304]:   Transition Length: 1.0s
[21:51:47][D][light:265]: 'Right Lamp - Base' Setting:
[21:51:47][D][light:278]:   Brightness: 100%
[21:51:47][D][light:287]:   Red=100%, Green=73%, Blue=24%
[21:51:47][D][light:290]:   White Value: 35%
[21:51:47][D][light:304]:   Transition Length: 1.0s
```
When pressing the "up" or "down" the following logs are shown:
```
[21:51:24][D][light:265]: 'Right Lamp - Base' Setting:
[21:51:24][D][light:278]:   Brightness: 78%
[21:51:24][D][light:304]:   Transition Length: 1.0s
[21:51:24][D][light:265]: 'Right Lamp - Main' Setting:
[21:51:24][D][light:278]:   Brightness: 78%
[21:51:24][D][light:304]:   Transition Length: 1.0s
```

So basically, the Esp doesn't get any colour change message and does it anyways (again, the Hue lamps in the group which get the same message change only their brightness).
I can't really say I understand this behaviour.

**Problem-relevant YAML-configuration entries:**
```yaml
light:
  - platform: neopixelbus
    type: GRBW
    variant: SK6812
    pin: GPIO03
    num_leds: 36
    name: "Right Lamp - Joint"
    id: led_lamp

```

**Logs (if applicable):**
See above

**Additional information and things you've tried:**
Changing the brightness via the HA frontend works like a charm
